Client/Frontend/Browser/URIFixup.swift

```diff
@@ -28,10 +28,16 @@ class URIFixup {
 
         // If there is a ".", prepend "http://" and try again. Since this
         // is strictly an "http://" URL, we also require a host.
-        if let url = NSURL(string: "http://\(trimmed)") where url.host != nil {
+        if let url = punycodedURL("http://\(escaped!)") where url.host != nil {
```

Comment:
Note that this will crash if `escaped` is `nil`. Instead, please replace the line above:
```
let escaped = ...
```
with this:
```
guard let escaped = ... else { return nil }
```